### PR TITLE
Persist combined datasets

### PR DESCRIPTION
**Pull request by ghop02**
Here's the first go at persisting combined datasets! 

Couple Key concepts this introduces:
 * DatasetPointer - essentially a data structure/file that points to a persisted dataset.  It stores some basic info needed to download/upload datasets (if the dataset is in s3) and or load it from disk.
 * DatasetTag - a tag for a dataset.  Right now this is just used to store pointers to latest/stable datasets. Right now there is no validation while promoting a dataset from latest to stable, but in the future there may be a more involved process.  
 * Github action workflow for kicking off a dataset update triggered after covid-data-public update workflow runs.

Currently I'm not actually saving the persisted dataset in the repo.  The ideal process I would like is to snapshot the dataset and save it in s3, then have the dataset pointer in the repo that lets you download the snapshot.  

Pros of storing in s3:
 * All previous datasets are kept around, so we can easily compare different versions/look back in history and see what we did at an earlier date.
 * Don't have to bloat covid-data-model git history with large csv file changes - they add up fairly quickly.
 * Caches download, so only need to download the ~30MB file once.

Cons:
 * CSV file isn't as immediately visible (could be advantages to having it in repo for visibility purposes)

### CODE REVIEW

File: libs/datasets/timeseries.py [hunk 1]
```diff
@@ -243,3 +250,20 @@ def summarize(self):
...
+
+    def to_csv(self, path: pathlib.Path):
+        """Persists timeseries to CSV.
+
+        Args:
+            path: Path to write to.
+        """
+        common_df.write_csv(self.data, path, structlog.get_logger())
+
+    @classmethod
```
[the remark this answers is not in the document]
Comment (ghop02) [reply libs/datasets/timeseries.py:262]: For this case, I prefer these sorts of loading methods to be `classmethods` - I've gotten into annoying scenarios where a class is loaded through subclasses and returns a different class than I'm expecting.

File: libs/datasets/latest_values_dataset.py [hunk 2]
```diff
@@ -189,3 +194,13 @@ def get_record_for_fips(self, fips) -> dict:
             return {}
 
         return row.iloc[0].to_dict()
+
+    def to_csv(self, path: pathlib.Path):
+        data = self.data.set_index(CommonFields.FIPS).replace({pd.NA: np.nan}).convert_dtypes()
```
Comment (TomGoBravo) [inline libs/datasets/latest_values_dataset.py:199]: add comment about why common_df doesn't work here.

File: libs/datasets/latest_values_dataset.py [hunk 3]
```diff
@@ -189,3 +194,13 @@ def get_record_for_fips(self, fips) -> dict:
...
+        data.to_csv(path, date_format="%Y-%m-%d", index=True, float_format="%.12g")
+
+    @classmethod
+    def load_csv(cls, path: pathlib.Path):
+
+        df = pd.read_csv(path, dtype={CommonFields.FIPS: str})
```
Comment (TomGoBravo) [inline libs/datasets/latest_values_dataset.py:205]: add comment about why common_df doesn't work here.

File: libs/datasets/combined_dataset_utils.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+            from head.  Generally used in testing to sidestep building entire dataset.
+        timeseries_dataset: Optionally specify a TimeseriesDataset to persist instead of building
+            from head.  Generally used in testing to sidestep building entire dataset.
+
+    Returns: Tuple of DatasetPointers to latest and timeseries datasets.
+    """
+    if not latest_dataset:
+        latest_dataset = combined_datasets.build_us_latest_with_all_fields(skip_cache=True)
+    latest_pointer = persist_dataset(latest_dataset, path_prefix)
+    latest_pointer.save(pointer_path_dir)
```
[the remark this answers is not in the document]
Comment (ghop02) [reply libs/datasets/combined_dataset_utils.py:87]: yeah that feels alright.  I think it makes more sense in a tagless world

**Pull request merged**